samples/read_aux.c

```diff
@@ -0,0 +1,210 @@
+    case 'c':
+        fprintf(fp, "%d", (int8_t)(idx > -1 ? bam_auxB2i(data, idx) : bam_aux2i(data)));    //signed 1 byte data; bam_auxB2i - from array or bam_aux2i - non array data
+        break;
+    case 'C':
+        fprintf(fp, "%u", (uint8_t)(idx > -1 ? bam_auxB2i(data, idx) : bam_aux2i(data)));   //unsigned 1 byte data
+        break;
+    case 's':
+        fprintf(fp, "%d", (int16_t)(idx > -1 ? bam_auxB2i(data, idx) : bam_aux2i(data)));   //signed 2 byte data
+        break;
+    case 'S':
+        fprintf(fp, "%u", (uint16_t)(idx > -1 ? bam_auxB2i(data, idx) : bam_aux2i(data)));  //unsigned 2 byte data
+        break;
+    case 'i':
+        fprintf(fp, "%d", (int32_t)(idx > -1 ? bam_auxB2i(data, idx) : bam_aux2i(data)));   //signed 4 byte data
+        break;
+    case 'I':
+        fprintf(fp, "%u", (uint32_t)(idx > -1 ? bam_auxB2i(data, idx) : bam_aux2i(data)));  //unsigned 4 byte data
+        break;
+    case 'f':
+    case 'd':
+        fprintf(fp, "%g", (float)(idx > -1 ? bam_auxB2f(data, idx) : bam_aux2f(data)));     //floating point data, 4 bytes
+        break;
+    case 'H':
+    case 'Z':
+        fprintf(fp, "%s", bam_aux2Z(data));                                                 //array of char or hex data
+        break;
+    case 'B':                                                                               //array of char/int/float
+        auxBcnt = bam_auxB_len(data);                                                       //length of array
+        auxBType = bam_aux_type(data + 1);                                                  //type of element in array
+        fprintf(fp, "%c", auxBType);
+        for (i = 0; i < auxBcnt; ++i) {                                                     //iterate the array
+            fprintf(fp, ",");
+            printauxdata(fp, auxBType, i, data);                                            //calling recurssively  with index to reuse a few lines
+        }
+        break;
+    default:
+        fprintf(stderr, "Invalid aux tag?\n");
+        return EXIT_FAILURE;
+        break;
+    }
+    return EXIT_SUCCESS;
+}
+
+/// main_demo - start of the demo
+/** @param argc - count of arguments
+ *  @param argv - pointer to array of arguments
+returns 1 on failure 0 on success
+*/
+int main(int argc, char *argv[])
+{
+    static const struct option lopts[] = {
+        {NULL, 0, NULL, 0}
+    };
+
+    const char *inname = NULL;
+    int c = 0, ret = EXIT_FAILURE, idx = -1;
+    sam_hdr_t *in_samhdr = NULL;
+    samFile *infile = NULL;
+
+    int ret_r = 0, i = 0, auxcnt = 0, auxBcnt = 0, cnt = 0;
+    bam1_t *bamdata = NULL;
+    uint8_t *data = NULL;
+
+    //read_aux [-n position] [-i infile] [<infile>]
+    while ((c = getopt_long(argc, argv, "i:n:", lopts, NULL)) >= 0)
+    {
+        switch (c)
+        {
+        case 'i':                           //in file name
+            inname = optarg;
+            break;
+        case 'n':                           //position
```

Comment:
The comments for read_bam.c apply here too.  It's not position, but the Nth record.  Again consider reporting the first N instead of Nth.

Reply:
updated.